SimPEG/DataMisfit.py

```diff
@@ -1,65 +1,49 @@
 from __future__ import print_function
 import numpy as np
-from SimPEG import Utils
-from SimPEG import Survey
+import properties
 
+from . import Utils
+from . import Survey
+from . import ObjectiveFunction
+from .Tests import checkDerivative
 
-class BaseDataMisfit(object):
-    """BaseDataMisfit
+
+class BaseDataMisfit(ObjectiveFunction.L2ObjectiveFunction):
+    """
+    BaseDataMisfit
 
         .. note::
 
-            You should inherit from this class to create your own data misfit term.
+            You should inherit from this class to create your own data misfit
+            term.
     """
 
     debug   = False  #: Print debugging information
-    counter = None   #: Set this to a SimPEG.Utils.Counter() if you want to count things
+    counter = None  #: Set this to a SimPEG.Utils.Counter() if you want to count things
+
+    _hasFields = True  #: Data Misfits take fields, handy to store them
```

Comment:
this is needed by the `BaseObjectiveFunction` so that it knows whether or not to expect fields when computing a residual and if fields should be passed (eg. regularization should not take in fields, but data misfits can)